tests/purs/warning/DuplicateExportRef.purs

```diff
@@ -5,6 +5,12 @@
 -- @shouldWarnWith DuplicateExportRef
 -- @shouldWarnWith DuplicateExportRef
 -- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
+-- @shouldWarnWith DuplicateExportRef
 module Main
   ( X(X, X), X
```

Comment:
Was wondering whether it makes sense to warn for all duplicates, or just the follow-on duplicates.
I believe there used to be two warnings for this line. Now there are three.

Reply:
Ah, right, yeah this looks like something that we will need to fix. At the moment lots of the warnings are duplicated:

```

Warning 1 of 13:

  in module [33mMain[0m
  at tests/purs/warning/DuplicateExportRef.purs:14:1 - 36:28 (line 14, column 1 - line 36, column 28)

    Export list contains multiple references to type [33mX[0m


  See https://github.com/purescript/documentation/blob/master/errors/DuplicateExportRef.md for more information,
  or to contribute content related to this warning.

Warning 2 of 13:

  in module [33mMain[0m
  at tests/purs/warning/DuplicateExportRef.purs:14:1 - 36:28 (line 14, column 1 - line 36, column 28)

    Export list contains multiple references to type [33mX[0m


  See https://github.com/purescript/documentation/blob/master/errors/DuplicateExportRef.md for more information,
  or to contribute content related to this warning.
```

I think since the warning text just says that the export list contains multiple references to a particular thing, then we should only have one warning per group of duplicates, i.e. we should have at most one warning that the export list contains multiple references to a particular type or type class or whatever. If we wanted to change the warning so that the source span pointed to a particular duplicated reference, and change the text a bit so that it said something like "The export list already contains a reference to [thing], so this reference is redundant," then I think it would be okay to emit one warning for each duplicate after the first occurrence. That's probably not worth doing in this PR, though, and also I think if we wanted to consider that approach we'd need to be careful about how it might affect editor plugins and other tools which can automatically fix this warning.

Reply:
Thinking it's best to just stick with the existing behavior for now, and let this PR stay more focused on the performance fix.